module/VuFind/sql/mysql.sql

```diff
@@ -35,14 +35,14 @@ CREATE TABLE `change_tracker` (
 /*!40101 SET character_set_client = utf8 */;
 CREATE TABLE `comments` (
   `id` int(11) NOT NULL AUTO_INCREMENT,
-  `user_id` int(11) NOT NULL DEFAULT '0',
+  `user_id` int(11) DEFAULT NULL,
   `resource_id` int(11) NOT NULL DEFAULT '0',
   `comment` text NOT NULL,
   `created` datetime NOT NULL DEFAULT '2000-01-01 00:00:00',
   PRIMARY KEY (`id`),
   KEY `user_id` (`user_id`),
   KEY `resource_id` (`resource_id`),
-  CONSTRAINT `comments_ibfk_1` FOREIGN KEY (`user_id`) REFERENCES `user` (`id`) ON DELETE CASCADE,
+  CONSTRAINT `comments_ibfk_1` FOREIGN KEY (`user_id`) REFERENCES `user` (`id`) ON DELETE SET NULL,
```

Comment:
I suspect that the current database upgrade script does not account for changed behavior on foreign keys (though it's worth testing in case I already implemented but forgot about it). If that's not working, we should figure out a way to make it work. I imagine you may not have much stomach for digging into that less-than-beautiful code, but it would at least be helpful to put a TODO checkbox on the PR so I can look into it for you.

Reply:
I'll check if it works and whether I can make it work if it doesn't.

Reply:
This is now in PR #1108.

Reply:
Thanks -- I've merged that to master, and from master to here.